beacon_node/store/src/hot_cold_store.rs

```diff
@@ -2116,39 +2117,27 @@ impl<E: EthSpec, Hot: ItemStore<E>, Cold: ItemStore<E>> HotColdDB<E, Hot, Cold>
     /// As for `compare_and_set_anchor_info`, but also writes the anchor to disk immediately.
     pub fn compare_and_set_anchor_info_with_write(
         &self,
-        prev_value: Option<AnchorInfo>,
-        new_value: Option<AnchorInfo>,
+        prev_value: AnchorInfo,
+        new_value: AnchorInfo,
     ) -> Result<(), Error> {
         let kv_store_op = self.compare_and_set_anchor_info(prev_value, new_value)?;
         self.hot_db.do_atomically(vec![kv_store_op])
     }
 
-    /// Load the anchor info from disk, but do not set `self.anchor_info`.
-    fn load_anchor_info(&self) -> Result<Option<AnchorInfo>, Error> {
-        self.hot_db.get(&ANCHOR_INFO_KEY)
+    /// Load the anchor info from disk.
+    fn load_anchor_info(hot_db: &Hot) -> Result<AnchorInfo, Error> {
+        // FIXME(sproul): add migration
```

Comment:
I still need to add this to the v22 DB migration:

- Set the anchor to `ANCHOR_FOR_ARCHIVE_NODE` if it is currently absent (`None`).